src/main.rs

```diff
@@ -528,6 +529,11 @@ fn notmain() -> Result<i32, anyhow::Error> {
     )
 }
 
+fn program_size_of(file: &ElfFile) -> u64 {
+    // `segments` iterates only over *loadable* segments, which are the segments that will be loaded to Flash by probe-rs
+    file.segments().map(|segment| segment.size()).sum()
```

Comment:
Stylistic nit: could just `.map(Segment::size)`

Reply:
(did not compile because `segments` yields `Segment` values but `Segment::size` expects a reference (`&Segment`))